service/src/main/kotlin/xyz/lebkuchenfm/domain/xsounds/XSoundsService.kt

```diff
@@ -13,11 +17,16 @@ class XSoundsService(private val repository: XSoundsRepository, private val file
         return repository.findAllByTagOrderByNameAsc(tag)
     }
 
-    suspend fun addNewXSound(soundName: String, tags: List<String>, bytes: ByteArray): XSound {
-        val fileUrl = fileRepository.uploadXSoundFile(soundName, bytes)
+    sealed interface NewSoundError {
+        data object CouldNotAddNewSound : NewSoundError
+    }
+
+    suspend fun addNewXSound(soundName: String, tags: List<String>, bytes: ByteArray): Result<XSound, NewSoundError> {
+        val fileUrl = fileRepository.uploadXSoundFile(soundName, bytes).get()
+            ?: return Err(NewSoundError.CouldNotAddNewSound)
         val readySound = XSound(name = soundName, url = fileUrl, tags = tags, timesPlayed = 0, addedBy = null)
         repository.insert(readySound)
```

Comment:
We should check if the sound with the same name already exists. This is another type of `NewSoundError`.